### PR TITLE
Cherry-pick 22 patches from `main` to `2.3.x`

### PR DESCRIPTION
**Pull request by cdisselkoen**
## Description of changes

Cherry-picks the following 22 PRs from `main` to `2.3.x`:
* #90 
* #91
* #93 
* #96 
* #102 
* #97 
* #92 
* #121 
* #122 
* #124 
* #131 
* #139 
* #138 
* #136 
* #151 
* #147 
* #158 
* #155 
* #173 
* #179 
* #192 
* #186 

Also bumps the version numbers of all packages to 2.3.3 (as 2.3.2 was already released).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by khieta — APPROVED** [review]
Looks good! Based on the description though, I'm not sure that #117 should be a patch. It seems more like a functionality change (minor). What do you think @john-h-kastner-aws?

**Comment by cdisselkoen** [comment]
Downstream build of `cedar-spec` is failing, perhaps because the CI is trying to use `main` of `cedar-spec`, rather than `2.3.x` of `cedar-spec`.  Is there already an issue for this?

**Comment by khieta** [comment]
Yes, see #229

**Comment by cdisselkoen** [comment]
I dropped #117.  I also had to drop #165 as it seems to depend on #117 (or at least, taking #165 without #117 is a nontrivial merge which we could do in a separate PR to `release/2.3.x` if necessary).

**Comment by cdisselkoen** [comment]
Also updated the PR title and description to reflect the corrected list of commits

**Comment by john-h-kastner-aws** [comment]
#165 is entirely dependent on #117, so that sounds good to me

File: cedar-policy-cli/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+### Changed
+- Input policies for `check-parse` command can be read from standard input.
```
Comment (john-h-kastner-aws) [inline cedar-policy-cli/CHANGELOG.md:6]: We'll probably take a separate look at change logs prior to release, but this change in particular isn't included in the release branch

**Review by john-h-kastner-aws — APPROVED** [review]
See comment here about PR #151. We need to call this change out for AVP. https://github.com/cedar-policy/cedar/issues/150#issuecomment-1621936143

**Pull request merged**